Sources/Filters/General/ClipClosedSurface/index.js

```diff
@@ -0,0 +1,1100 @@
+  }
+
+  // function breakTriangleStrips(inputStrips, polys, inputScalars, firstStripScalar, polyScalars, color) {
+  //   if (!inputStrips) {
+  //     return;
+  //   }
+
+  //   // TODO
+  //   let npts = 0;
+  //   const vtkIdType* pts = null;
+
+  //   inputStrips.InitTraversal();
+
+  //   for (let cellId = firstStripScalar; inputStrips.getNextCell(npts, pts); cellId++) {
+  //     vtkTriangleStrip::DecomposeStrip(npts, pts, polys);
+  //   // TODO: Support of TriangleStrips is optional/nice to have
+
+  //     if (polyScalars) {
+  //       let scalarValue = [...color];
+
+  //       if (inputScalars) {
+  //         // If there are input scalars, use them instead of "color"
+  //         inputScalars.getTypedTuple(cellId, scalarValue);
+  //       }
+
+  //       let n = npts - 3;
+  //       let m = polyScalars.getNumberOfTuples();
+  //       if (n >= 0) {
+  //         // First insert is just to allocate space
+  //         // polyScalars.insertTypedTuple(m + n, scalarValue);
+
+  //         for (let i = 0; i < n; i++) {
+  //           // polyScalars.setTypedTuple(m + i, scalarValue);
+  //           polyScalars.setTuple(m + i, scalarValue);
+  //         }
+  //       }
+  //     }
+  //   }
+  // }
+
+  function triangulateContours(
+    polyData,
+    firstLine,
+    numLines,
+    outputPolys,
+    normal
+  ) {
+    // If no cut lines were generated, there's nothing to do
+    if (numLines <= 0) {
+      return;
+    }
+
+    const nnormal = [-normal[0], -normal[1], -normal[2]];
+    const rval = vtkContourTriangulator.triangulateContours(
+      polyData,
+      firstLine,
+      numLines,
+      outputPolys,
+      nnormal
+    );
+
+    if (rval === 0 && model.triangulationErrorDisplay) {
+      vtkErrorMacro('Triangulation failed, polyData may not be watertight.');
+    }
+  }
+
+  function triangulatePolygon(polygon, points, triangles) {
+    return vtkContourTriangulator.triangulatePolygon(
+      polygon,
+      points,
+      triangles
+    );
+  }
+
+  function clipAndContourPolys(
+    points,
+    pointScalars,
+    pointData,
+    edgeLocator,
+    triangulate,
+    inputCells,
+    outputPolys,
+    outputLines,
+    inCellData,
+    outPolyData,
+    outLineData
+  ) {
+    const idList = model._idList;
+    // How many sides for output polygons?
+    // let polyMax = VTK_INT_MAX;
+    let polyMax = Number.MAX_VALUE;
+    if (triangulate) {
+      if (triangulate < 4) {
+        // triangles only
+        polyMax = 3;
+      } else if (triangulate === 4) {
+        // allow triangles and quads
+        polyMax = 4;
+      }
+    }
+
+    let triangulationFailure = 0;
+
+    // Go through all cells and clip them
+    // TODO: Check
+    const values = inputCells.getData();
+    const linePts = [0, 0];
+    let cellId = 0;
+    let numPts;
+    for (let i = 0; i < values.length; i += numPts + 1, cellId++) {
+      numPts = values[i];
+      let i1 = values[i + numPts];
+      let v1 = pointScalars.getValue(i1);
```

Comment:
different than  line 208. (I'm not sure it is a good idea to expose `getValue()`, pointsScalars.getData()[i1] should be enough).